app/main.py

```diff
@@ -1 +1,34 @@
-# write your code here
+from __future__ import annotations
+
+
+class Animal:
+    alive = []
+
+    def __init__(self,
+                 name: str,
+                 health: int = 100,
+                 hidden: bool = False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    @staticmethod
+    def bite(victim: Herbivore | Carnivore) -> None:
```

Comment:
fix type annotation here, Carnivore can't be bitten 
![image](https://user-images.githubusercontent.com/103201010/218459664-5920f6ab-bafa-4c92-95cc-246913428dbc.png)